Source/Classes/PINProgressiveImage.m

```diff
@@ -116,9 +117,14 @@ - (CFTimeInterval)startTime
     return startTime;
 }
 
-- (void)updateProgressiveImageWithData:(NSData *)data expectedNumberOfBytes:(int64_t)expectedNumberOfBytes
+- (void)updateProgressiveImageWithData:(nonnull NSData *)data expectedNumberOfBytes:(int64_t)expectedNumberOfBytes isResume:(BOOL)isResume
 {
     [self.lock lock];
+        if (isResume) {
+            NSAssert(self.mutableData == nil, @"If we're resuming, data shouldn't be setup yet.");
+            self.startingBytes = data.length;
```

Comment:
We should build some magic debug macro that adds in code on DEBUG which automatically checks that locks are held when a method prefixed with _locked_ is called!